File: Stage22/C/Problem16139.c
#include <stdio.h>
#include <string.h>

int prefix[26][200001];

int main(void) {
    char S[200001];
    scanf("%s", S);

    int n = strlen(S);

    for (int i = 0; i < n; i++) {
        int c = S[i] - 'a';

        for (int j = 0; j < 26; j++) {
            prefix[j][i + 1] = prefix[j][i];
        }

        prefix[c][i + 1]++;
    }

    int q;
    scanf("%d", &q);

    while(q--) {
        char ch;
        int l, r;
        scanf(" %c %d %d", &ch, &l, &r);

        int idx = ch - 'a';
        printf("%d\n", prefix[idx][r + 1] - prefix[idx][l]);
    }

    return 0;
}

File: Stage22/Java/Problem16139.java
package Stage22.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Problem16139 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        String S = br.readLine();
        int n = S.length();

        int[][] prefix = new int[26][n];

        for (int i = 0; i < n; i++) {
            int c = S.charAt(i) - 'a';

            for (int j = 0; j < 26; j++) {
                prefix[j][i] = (i > 0) ? prefix[j][i - 1] : 0;
            }
            prefix[c][i]++;
        }
        
        int q = Integer.parseInt(br.readLine());

        for (int i = 0; i < q; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            char ch = st.nextToken().charAt(0);

            int l = Integer.parseInt(st.nextToken());
            int r = Integer.parseInt(st.nextToken());

            int idx = ch - 'a';

            if (l == 0) {
                sb.append(prefix[idx][r]);
            } else {
                sb.append(prefix[idx][r] - prefix[idx][l - 1]);
            }

            sb.append('\n');
        }

        System.out.println(sb);
        br.close();
    }
}
